waiter/integration/waiter/killed_instance_test.clj

```diff
@@ -65,50 +43,73 @@
-                other-request-instances (atom {})]
-            (bombard-with-requests-fn 20 400 other-request-instances)
-            (log/info "checking whether killed instance was subsequently reserved.")
-            (is (not (contains? @other-request-instances killed-instance-id1))
-                (str killed-instance-id1 "was used to service requests" (get @other-request-instances killed-instance-id1)))
-            (is (not (contains? @other-request-instances killed-instance-id2))
-                (str killed-instance-id2 "was used to service requests" (get @other-request-instances killed-instance-id2)))
-            (async/<!! blacklist-check-chan))
-          (log/warn "skipping assertions as only" num-tasks-running "instances running for" service-id)))
-      (delete-service waiter-url service-id)
-      (log/info "testing blacklisted instance reservation avoided completed."))))
+    (log/info "Testing blacklisted instance is not reserved")
+    (let [router-id->router-url (routers waiter-url)
+          num-routers (count router-id->router-url)
+          extra-headers {:x-waiter-blacklist-on-503 true
+                         :x-waiter-concurrency-level (inc num-routers)
+                         :x-waiter-name (rand-name)
+                         :x-waiter-scale-up-factor 0.99}
+          make-request-fn (fn make-request-fn []
+                            (make-request-with-debug-info
+                              extra-headers #(make-kitchen-request waiter-url %)))
+          {:keys [cookies request-headers instance-id service-id] :as response} (make-request-fn)]
+      (assert-response-status response 200)
+      (log/info "canary instance-id:" instance-id)
+      (with-service-cleanup
+        service-id
+        (dotimes [_ 4] ;; incrementally cause longer blacklist duration for instance
+          (->> router-id->router-url
+               (map (fn [[_ router-url]]
+                      (launch-thread
+                        (trigger-blacklisting-of-instance router-url request-headers cookies))))
+               (map async/<!!)
+               doall))
+
+        (let [router-id->blacklist-time
+              (pc/map-from-keys
+                (fn [router-id]
+                  (let [router-url (router-id->router-url router-id)]
+                    (-> #(instance-blacklisted-by-router? router-url service-id instance-id cookies)
+                        (wait-for :interval 5 :timeout 10))))
+                (keys router-id->router-url))]
+          (doseq [[router-id _] router-id->router-url]
+            (is (router-id->blacklist-time router-id)
+                (str "instance not blacklisted " {:instance-id instance-id :router-id router-id})))
+
+          (let [new-request-instance-ids-atom (atom #{})]
+            (parallelize-requests
+              10 1
+              #(let [{:keys [instance-id router-id]} (make-request-fn)]
```

Comment:
I'd rather use `fn` here, I kept looking for the `%` inside the code.

Reply:
Okay, will do.

Reply:
Done.